paimon-core/src/main/java/org/apache/paimon/sort/zorder/ZIndexer.java

```diff
@@ -0,0 +1,373 @@
+        }
+
+        @Override
+        public ZorderBaseFunction visit(BooleanType booleanType) {
+            return new ZorderBooleanFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(BinaryType binaryType) {
+            return new ZorderBytesFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(VarBinaryType varBinaryType) {
+            return new ZorderBytesFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(DecimalType decimalType) {
+            return new ZorderDecimalFunction(decimalType.getPrecision(), decimalType.getScale());
+        }
+
+        @Override
+        public ZorderBaseFunction visit(TinyIntType tinyIntType) {
+            return new ZorderTinyIntFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(SmallIntType smallIntType) {
+            return new ZorderShortFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(IntType intType) {
+            return new ZorderIntFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(BigIntType bigIntType) {
+            return new ZorderLongFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(FloatType floatType) {
+            return new ZorderFloatFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(DoubleType doubleType) {
+            return new ZorderDoubleFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(DateType dateType) {
+            return new ZorderDateFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(TimeType timeType) {
+            return new ZorderTimeFunction();
+        }
+
+        @Override
+        public ZorderBaseFunction visit(TimestampType timestampType) {
+            return new ZorderTimestampFunction(timestampType.getPrecision());
+        }
+
+        @Override
+        public ZorderBaseFunction visit(LocalZonedTimestampType localZonedTimestampType) {
+            return new ZorderTimestampFunction(localZonedTimestampType.getPrecision());
+        }
+
+        @Override
+        public ZorderBaseFunction visit(ArrayType arrayType) {
+            throw new RuntimeException("Unsupported type");
+        }
+
+        @Override
+        public ZorderBaseFunction visit(MultisetType multisetType) {
+            throw new RuntimeException("Unsupported type");
+        }
+
+        @Override
+        public ZorderBaseFunction visit(MapType mapType) {
+            throw new RuntimeException("Unsupported type");
+        }
+
+        @Override
+        public ZorderBaseFunction visit(RowType rowType) {
+            throw new RuntimeException("Unsupported type");
+        }
+    }
+
+    /** BaseFunction to convert row field record to devoted bytes. */
+    public abstract static class ZorderBaseFunction
```

Comment:
You can refer to `InternalRow.createFieldGetter`

Reply:
done